app/controllers/api/base_controller/parameters.rb

```diff
@@ -100,6 +102,8 @@ def parse_filter(filter, operators)
                         end
                       end
 
+        filter_value = from_cid(filter_value) if filter_attr =~ /[_]?id$/ && cid?(filter_value)
```

Comment:
they are optional, I tend to use them regardless, one or more character being matched for consistency.